src/LanguageServer/Impl/Implementation/WorkspaceManager.cs

```diff
@@ -0,0 +1,74 @@
+﻿using Microsoft.Extensions.FileSystemGlobbing;
+using Microsoft.Extensions.FileSystemGlobbing.Abstractions;
+using Microsoft.PythonTools.Analysis;
+using Microsoft.PythonTools.Analysis.Infrastructure;
+using Microsoft.PythonTools.Parsing;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.Python.LanguageServer.Implementation {
+    class WorkspaceManager {
```

Comment:
`internal sealed class`. `WorkspaceManager` sounds a bit generic. Since it is specific to the indexer, you may want to add separate `Indexing` folder with namespace like `Microsoft.Python.LanguageServer.Indexing` and perhaps name `WorkspaceIndexingManager`.